org.eclipse.sisu.plexus/src/org/codehaus/plexus/component/configurator/converters/basic/PathConverter.java

```diff
@@ -0,0 +1,57 @@
+/*******************************************************************************
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Minimal facade required to be binary-compatible with legacy Plexus API
+ *******************************************************************************/
+package org.codehaus.plexus.component.configurator.converters.basic;
+
+import java.io.File;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import org.codehaus.plexus.component.configurator.ComponentConfigurationException;
+import org.codehaus.plexus.component.configurator.ConfigurationListener;
+import org.codehaus.plexus.component.configurator.converters.lookup.ConverterLookup;
+import org.codehaus.plexus.component.configurator.expression.ExpressionEvaluator;
+import org.codehaus.plexus.configuration.PlexusConfiguration;
+
+public class PathConverter
```

Comment:
does it make sense to delegate to a FileConverter internal instance and just convert the `File` to `Path` using `toPath()` after? would avoid to duplicate the logic even if not very complex

Can also be worth testing both codepaths IMHO.

Reply:
IMHO testing both File and Path (both relative and absolute) is covered in https://github.com/eclipse/sisu.plexus/pull/22/files#diff-89f93149e855745f1ef9701ca4c2ae8a748803ee87334873a0197780981939fbR49.

Purely doing delegation from PathConverter to FileConverter feels like the wrong way around, as certain edge cases (e.g. different fileystems with potentially different name separators) are only supported by `java.nio.file.Path`). If at all I would rather delegate from FileConverter to PathConverter.

Unfortunately I cannot extend the interface `ExpressionEvaluator` in a backwards compatible way, as that would require default methods introduced with Java 8.

Reply:
Well, it is more about typing, value should be close and File handles better OS than Path (Path.of/Paths.get does not work well on absolute win path whereas File does) so it is close. Only advantage of Path is where it is not about local filesystem and it sounds out of scope of this converter since it uses a root/basedir dir so sounds ok to me to do it

Reply:
Can you elaborate on 
> Path.of/Paths.get does not work well on absolute win path whereas File does

If that is the case, that pretty much sounds like a Java bug to me. With which Java version did you experience those issues?

Reply:
Java 8, 11 or 17 (ea for this one, didnt test recently).
Often you get paths starting with a slash (coming from url or other locations so something like `/c:/....`). Works well and "valid" with File, broken with Path but file to path mapping works.
Not sure why both dont have the same behavior for the same local filesystem - one should be backed by the other no? ;) - but it is easy to get trapped there injecting a computed path (interpolation/mojo computed property in mvn land).
Going through File avoids this pitfall but my comment was mainly to ensure we get a single time the handling of paths whatever type it is.